First task

farkaszoli opened the pull request.

farkaszoli committed:
first task

farkaszoli committed:
Task2

v-tgriz commented:
.idea/uiDesigner.xml
out/production/Practice/Main.class
Ezeket nem kellene gitbe rakni.
Mindig ellenőrizd le a módosított file-okat commit előtt!

v-tgriz commented on src/KutyasFeladat.java:
```diff
@@ -0,0 +1,48 @@
+
+        for(Kutya kutya : kutyaList)
+        {
+            if(!kutya.getFajta().equals("vizsla"))
```
"vizsla".equals(kutya.getFajta())
Így az equals nem szalad NPE-re, ha a fajta nincs kitöltve. (Ha a kutya null persze még mindig bukik.)
Ez konvenció, mindig a string literálhoz, konstanshoz hasonlítunk.

v-tgriz commented on src/Practice.java:
```diff
@@ -0,0 +1,72 @@
+
+            if (name.charAt(i) == 'a') {
+                System.out.print("contains");
+            }
```
Ez két feladat összevonása. A második feladat kíírásának nem tesz eleget.

v-tgriz commented on src/Practice.java:
```diff
@@ -0,0 +1,72 @@
+            }
+        }
+        
+        String[] lastName = name.split(" ");
```
Félrevezető változónév.

v-tgriz commented on src/Practice.java:
```diff
@@ -0,0 +1,72 @@
+        
+        String[] lastName = name.split(" ");
+        System.out.print(lastName[0]);
+        System.out.print(lastName[1]);
```
Szerencsésebb lett volna, ha megjelölöd, hogy melyiket írod ki éppen.

v-tgriz commented on src/Practice.java:
```diff
@@ -0,0 +1,72 @@
+        System.out.println(lastName[0].length());
+        
+        System.out.println(lastName[1].length());
+    }
```
A fentebbi komplett feladatsor kiírva a képernyőre nem úgy fog kinézni, ahogy kellene.

v-tgriz commented on src/Practice.java:
```diff
@@ -0,0 +1,72 @@
+            System.out.print(taskWord.charAt(i));
+        }
+
+        System.out.print("\n");
```
println()

v-tgriz commented on src/Practice.java:
```diff
@@ -0,0 +1,72 @@
+        String[] words = taskWord.split(" ");
+        for (String word : words)
+        {
+            System.out.print(word + "\n");
```
println()

v-tgriz commented on src/StringCounter.java:
```diff
@@ -0,0 +1,27 @@
+public class StringCounter
+{
+    private String fromString = "KKNDDKDKKNKDEDNNENNNNNEKNEEEDKNK"+
```
Ezeket a stringeket tedd konstanssá

v-tgriz commented on src/Egypercesek.java:
```diff
@@ -0,0 +1,24 @@
+     */
+
+    public void firstCharIsCapital() {
+        String text = "ez egy mondat lenne, melynek minden szavat nagy kezdobetusse kellene alakitani.";
```
Ha ez egy konstans string, akkor kezeld is úgy! Emeld ki!

v-tgriz commented on src/StringCounter.java:
```diff
@@ -0,0 +1,27 @@
+    public void counter()
+    {
+        int searchStringLength = searchString.length();
+        int fromStringLength = fromString.length();
```
Ezt a változót egyetlen helyen használod, ezért nem érdemes kiemelni a műveletet.

v-tgriz commented:
a plant.csv nem tartozik egyik feladathoz sem

farkaszoli committed:
refactor project
- add package structure
- code review changes
- add new tasks
- extend gitignore file
- remove unused cs file

taz1988 commented:
töröld ki azokat a fájlokat amik nem tartoznak ide (.idea, *.class, stb.). Ha nem megy akkor szólj aztán megnézzük együtt.

farkaszoli committed:
Novenyes feladat
